### PR TITLE
#2794 fix document attachment label

### LINKED ISSUE #2794: Attaching document in chat shows a default name of "Chat_attachement" instead of the document name
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
Preview of the document should show the name of the file. 

## Actual Result:
Default name is displayed "chat_attachment"

## Action Performed:
1. Login in chat app. 
2. Go to a conversation 
3. Tap on the attachment icon
4. Tap on document
5. Select a document. 

## Workaround:
No need, visual issue. 

## Platform:
Where is this issue occurring?

 Web
 iOS
 Android ✔️ 
 Desktop App
 Mobile Web

**Version Number:** 1.0.41-11

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** 

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___
From @quinthar https://expensify.slack.com/archives/C01GTK53T8Q/p1620702841036200
> tapping + > Add Attachment > Choose Document on Android native brings up a weird preview, but ultimately works.  The preview should show what was picked.

view the job here: https://www.upwork.com/jobs/~0140aec4e5587952a8

### PR DESCRIPTION
**Pull request by dklymenk**
### Details
Documentation for [RNDocumentPicker](https://github.com/rnmods/react-native-document-picker#name) says that the returned object has `name` and not `fileName` key.

### Fixed Issues

Fixes #2794 

### Tests/QA Steps

PDF attachment:
1. Login in chat app.
2. Go to a conversation
3. Tap on the attachment icon
4. Tap on document
5. Select a PDF document.
6. You should see a preview

non-PDF non-image attachment:
1. Login in chat app.
2. Go to a conversation
3. Tap on the attachment icon
4. Tap on document
5. Select a non-PDF non-image file (I used `.xsl`).
6. You should see a preview with file name only.

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [x] iOS
- [x] Android

##### PDF
Simulator Screen Shot - iPhone 11 - 2021-05-14 at 03 06 37

##### non-PDF non-image
Simulator Screen Shot - iPhone 11 - 2021-05-14 at 03 07 28

##### PDF
Screenshot_1620951012

##### non-PDF non-image
Screenshot_1620951078

### CODE REVIEW

**Comment by dklymenk** [comment]
The diff affects only one `.native.js`, so to my understanding only iOS and Android should be tested.

Please let me know if I should add screenshots from other clients.

**Comment by chiragsalian** [comment]
Bump @thienlnam to review.

**Comment by thienlnam** [comment]
Tested IOS last friday (Worked well), currently running into issues with my android emulator so I will merge once I get that sorted out and tested

**Review by thienlnam — APPROVED** [review]
Nice, works as expected!

File: src/components/AttachmentPicker/index.native.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const documentPickerOptions = {
  */
 function getDataForUpload(fileData) {
     return {
-        name: fileData.fileName || 'chat_attachment',
+        name: fileData.name || 'chat_attachment',
```
Comment (kidroca) [inline src/components/AttachmentPicker/index.native.js:49]: This fixes the issue for the document picker but breaks it for the image picker as both use this function
one would call it with `fileData` that has `fileName` - image-picker results, the other with `fileData` that has just `name` 
This should have been changed to 

```js
name: fileData.fileName || fileData.name || 'chat_attachment'
```

Comment (thienlnam) [reply src/components/AttachmentPicker/index.native.js:49]: I'd recommend creating a new pull request with those changes, and tag the new issue that @kidroca created and then add me as a reviewer on it

Comment (dklymenk) [reply src/components/AttachmentPicker/index.native.js:49]: @thienlnam 
Here is the PR. I can't manually add you to the list of reviewers.
https://github.com/Expensify/Expensify.cash/pull/3408

Thanks

**Pull request merged**